JsonPreprocessor/CJsonPreprocessor.py

```diff
@@ -684,15 +684,18 @@ def __handleStrNoneTrueFalse(objJson):
                 i=0
                 for item in items:
                     i+=1
-                    subItems = re.split("\s*,\s*", item)
                     newSubItem = ""
-                    j=0
-                    for subItem in subItems:
-                        j+=1
-                        if j<len(subItems):
-                            newSubItem = newSubItem + self.__checkAndUpdateKeyValue(subItem) + ", "
-                        else:
-                            newSubItem = newSubItem + self.__checkAndUpdateKeyValue(subItem)
+                    if re.search("\s*\[[^\[\]]*\]\s*,*\s*", item):
```

Comment:
Hi Son,
can you explain me the code a little.
I would expect that inside a string (means between double quotes) data must be only processed if a `${` sequence exists. Otherwise it mustn't be touched.
Thank you,
 Thomas

Reply:
Hello Thomas,

This code I handle for the situation that a value (right hand side of ":" in Json config file) is a list and this list contains nested parameters. For example, "key" : [1, "two", ${nested}['abc']['def'], 3, "four"] then the value of nested param will be replaced in this list.

I think I forgot the case that a list value could be separated into multiple lines.

I will update my change tomorrow.

Thank you,
Son